bemani/backend/ddr/base.py

```diff
@@ -142,6 +149,24 @@ def unformat_profile(self, userid: UserID, request: Node, oldprofile: Profile) -
         """
         return oldprofile
 
+    def get_profile_by_refid_and_part(self, refid: Optional[str], part: Optional[str]) -> Optional[Node]:
```

Comment:
Yeah, this definitely feels weird. But, the pattern is established. Worth a think.